pkg/kubelet/cm/cgroup_manager_linux.go

```diff
@@ -388,6 +404,62 @@ func getCpuMax(cpuQuota *int64, cpuPeriod *uint64) string {
 	return fmt.Sprintf("%s %s", quotaStr, periodStr)
 }
 
+// propagateControllers on an unified hierarchy enables all the supported controllers for the specified cgroup
+func propagateControllers(path string) error {
+	if err := os.MkdirAll(path, 0755); err != nil {
+		return fmt.Errorf("failed to create cgroup %q : %v", path, err)
+	}
+
+	// Retrieve all the supported controllers from the cgroup root
+	controllersFileContent, err := ioutil.ReadFile(filepath.Join(cmutil.CgroupRoot, "cgroup.controllers"))
+	if err != nil {
+		return fmt.Errorf("failed to read controllers from %q : %v", cmutil.CgroupRoot, err)
+	}
+	// The retrieved content looks like: "cpuset cpu io memory hugetlb pids".  Prepend each of the controllers
```

Comment:
this appears to blindly take all the root controllers that were enabled on the system and propagates it to the children.  shouldn't we only propagate the controllers that are supported by k8s itself?  this feels like the place we should continue to maintain a whitelist of supported controllers.

Reply:
good idea.  I will change the test to only enable the controllers we need.

Should `pids` be behind the feature gate?

Reply:
thanks, done